pkg/utils/utils.go

```diff
@@ -904,3 +908,31 @@ func IsObjectBlocked(item runtime.Unstructured) (bool, string, error) {
 	}
 	return false, crdName, nil
 }
+
+func GetBackupdriverClient(config *rest.Config) (v1alpha1.BackupdriverV1alpha1Interface, error) {
+	pluginClient, err := plugin_clientset.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return pluginClient.BackupdriverV1alpha1(), nil
+}
+
+// Provide a utility function in guest cluster to clean up corresponding supervisor cluster snapshot CR
+func DeleteSvcSnapshot(gcSnapshot *backupdriverapi.Snapshot, svcConfig *rest.Config, svcNamespace string, logger logrus.FieldLogger) error {
+	svcSnapshotName := gcSnapshot.Status.SvcSnapshotName
```

Comment:
Add a check here to see if gcSnapshot.Status.SvcSnapshotName is an empty string.  If it is empty string, return here.